selftests/functional/test_skiptests.py

```diff
@@ -194,94 +194,78 @@ def tearDown(self):
-        self.class_module = script.Script(class_path,
-                                          AVOCADO_TEST_SKIP_CLASS_DECORATORS)
-        self.class_module.save()
-
-        class_if_path = os.path.join(self.tmpdir.name,
-                                     'test_skip_if_class_decorators.py')
-        self.class_if_module = script.Script(class_if_path,
-                                             AVOCADO_TEST_SKIP_IF_CLASS_DECORATORS)
-        self.class_if_module.save()
-
-        class_unless_path = os.path.join(self.tmpdir.name,
-                                         'test_skip_unless_class_decorators.py')
-        self.class_unless_module = script.Script(class_unless_path,
-                                                 AVOCADO_TEST_SKIP_UNLESS_CLASS_DECORATORS)
-        self.class_unless_module.save()
-
-        lib_path = os.path.join(self.tmpdir.name, 'lib_skip_decorators.py')
-        self.test_lib = script.Script(lib_path, AVOCADO_TEST_SKIP_LIB)
-        self.test_lib.save()
-
-        skip_setup_path = os.path.join(self.tmpdir.name,
-                                       'test_skip_decorator_setup.py')
-        self.skip_setup = script.Script(skip_setup_path,
-                                        AVOCADO_SKIP_DECORATOR_SETUP)
-        self.skip_setup.save()
-
-        bad_teardown_path = os.path.join(self.tmpdir.name,
-                                         'test_skip_decorator_teardown.py')
-        self.bad_teardown = script.Script(bad_teardown_path,
-                                          AVOCADO_SKIP_DECORATOR_TEARDOWN)
-        self.bad_teardown.save()
+    def _create_tmp_file(self, name, content):
+        scr_obj = script.Script(os.path.join(self.tmpdir.name, name), content)
+        scr_obj.save()
+        return scr_obj
 
-    def test_skip_decorators(self):
+    def check_skips(self, number_of_skips):
         cmd_line = [AVOCADO,
                     'run',
                     '--disable-sysinfo',
                     '--job-results-dir',
                     '%s' % self.tmpdir.name,
-                    '%s' % self.test_module,
+                    '%s' % self.script_to_exec,
                     '--json -']
         result = process.run(' '.join(cmd_line), ignore_status=True)
         json_results = json.loads(result.stdout_text)
         debuglog = json_results['debuglog']
 
         self.assertEqual(result.exit_status, exit_codes.AVOCADO_ALL_OK)
-        self.assertEqual(json_results['skip'], 5)
+        self.assertEqual(json_results['skip'], number_of_skips)
         debuglog_contents = genio.read_file(debuglog)
         self.assertFalse('setup executed' in debuglog_contents)
         self.assertFalse('test executed' in debuglog_contents)
         self.assertFalse('teardown executed' in debuglog_contents)
 
+
+class Skip(Base):
+
+    def setUp(self):
+        super(Skip, self).setUp()
+        _ = self._create_tmp_file('lib_skip_decorators.py',
```

Comment:
Is this assignment needed?

Reply:
pylint tends to not like discarding values from functions/methods that return them.